### ISSUE TITLE
PPC64 [native]: testcase test-spaces/SizeOfInitialHeap failing

### ISSUE THREAD

**Issue by jbajwa** [ibmruntimes/v8#20]
```
> out/ppc64.release/cctest --random-seed=-795785457 test-spaces/SizeOfInitialHeap --nohard-abort --nodead-code-elimination --nofold-constants

stderr:
#
# Fatal error in ../test/cctest/heap/test-spaces.cc, line 520
# Check failed: (heap->paged_space(i)->CommittedMemory())<(kMaxInitialSizePerSpace).
#

==== C stack trace ===============================

    out/ppc64.release/cctest() [0x106339b4]
    out/ppc64.release/cctest() [0x1062ce4c]
    out/ppc64.release/cctest() [0x102dec58]
    out/ppc64.release/cctest() [0x1025e604]
    out/ppc64.release/cctest() [0x100e84e4]
    /lib64/power8/libc.so.6(+0x24580) [0x3fffac874580]
    /lib64/power8/libc.so.6(__libc_start_main+0xc4) [0x3fffac874774]
```

Its PPC64 specific so it might be related to the OS page size being 64K.

**Comment by jbajwa** [comment]
git bisect just completed. First bad commit is https://codereview.chromium.org/2438683005
Quick look doesn't point to why it would fail on ppc64 (related OS page size).
Going to investigate further.

**Comment by zyt33** [comment]
Debug build error message:
```
#
# Fatal error in ../test/cctest/heap/test-spaces.cc, line 520
# Check failed: heap->paged_space(i)->CommittedMemory() < kMaxInitialSizePerSpace (2424832 vs. 2097152).
#
```

The error is consistant (the figures are the same.)

**Comment by zyt33** [comment]
The page size being 64K caused the system unable to free some small amount of memory in order to free up the memory so that the code space usage is below 2MB.

**Comment by zyt33** [comment]
In Debug mode it is supposed to skip check for the code space but the test did not skip that. Investigating.

**Comment by jbajwa** [comment]
@zyt33 That might not be related to this issue, lets upstream the fix for this testcase and we can create a separate CL for the debug mode issue.

**Comment by jbajwa** [comment]
also I'm guessing that debug mode issue is not arch-specific?

**Comment by jbajwa** [comment]
https://codereview.chromium.org/2523293002/ (waiting to get reviewed)
Thanks @zyt33 !

**Comment by jbajwa** [comment]
fix upstreamed to master and backported to 5.6 (https://codereview.chromium.org/2530233002/)
